inc/ustream.h

```diff
@@ -101,6 +101,34 @@ MOCKABLE_FUNCTION(, AZ_ULIB_RESULT, az_ustream_concat,
         AZ_USTREAM_MULTI_DATA_CB*, multi_data,
         AZ_RELEASE_CALLBACK, multi_data_release);
 
+/**
+  * @brief   Split a ustream at a given position.
+  *
+  *  The split will divide a given ustream into two parts, divided at a passed position. The result of this operation will be two
+  *     ustream instances. The first and original ustream will contain the data from the original position up to but not including 
+  *     the passed <tt>split_pos</tt>. The second ustream will contain the data from the passed <tt>split_pos</tt> up to the end of the
+  *     <tt>ustream_instance</tt> before the function was called. The <tt>split_pos</tt> should be relative to the position returned
+  *     from az_ustream_get_position() and the remaining size of <tt>ustream_instance</tt>.
+  *
+  * @param[in,out]      ustream_instance        The {@link AZ_USTREAM}* with the interface of 
+  *                                             the ustream. It cannot be <tt>NULL</tt>, and it shall be a valid ustream.
+  * @param[out]         ustream_instance_split  The {@link AZ_USTREAM}* with the interface of the ustream which will contain the 
+  *                                             controls to the data after <tt>split_pos</tt>. It cannot be <tt>NULL</tt>.
+  * @param[in]          split_pos               The position at which to split the passed <tt>ustream_instance</tt>. It cannot
```

Comment:
>The [](start = 48, length = 3)

Consider `The <tt>offset_t</tt> with the`

Reply:
ah yes the type should come first thank you

---
In reply to: [333760448](https://github.com/Azure/azure-ulib-c/pull/36#discussion_r333760448) [](ancestors = 333760448)